src/fiatExchanges/ExchangeQR.tsx

```diff
@@ -0,0 +1,191 @@
+import { RouteProp } from '@react-navigation/native'
+import { StackParamList } from 'src/navigator/types'
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import React, { useRef, useState, useLayoutEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, View, Text } from 'react-native'
+import { useDispatch } from 'react-redux'
+import { Screens } from 'src/navigator/Screens'
+import { StyledQRCodeDisplay } from 'src/qrcode/QRCode'
+import { useSelector } from 'react-redux'
+import { SVG, shareQRCode } from 'src/send/actions'
+import Logger from 'src/utils/Logger'
+import { FiatExchangeEvents } from 'src/analytics/Events'
+import BackButton from 'src/components/BackButton'
+import { emptyHeader } from 'src/navigator/Headers'
+import { CICOFlow } from './utils'
+import i18n from 'src/i18n'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import Share from 'src/icons/Share'
+import { currentAccountSelector } from 'src/web3/selectors'
+import { nameSelector } from 'src/account/selectors'
+import variables from 'src/styles/variables'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import Button from 'src/components/Button'
+import Paste from 'src/icons/Paste'
+import Clipboard from '@react-native-clipboard/clipboard'
+import ExchangesBottomSheet from 'src/components/ExchangesBottomSheet'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { ExternalExchangeProvider } from 'src/fiatExchanges/ExternalExchanges'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.ExchangeQR>
+
+export default function ExchangeQR({ route, navigation }: Props) {
+  const qrSvgRef = useRef<SVG>()
+  const { flow, exchanges } = route.params
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const address = useSelector(currentAccountSelector)
+  const displayName = useSelector(nameSelector)
+
+  const [bottomSheetVisible, setBottomSheetVisible] = useState(false)
+
+  const onCloseBottomSheet = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.external_exchange_qr_bottom_sheet_close, {
+      flow,
+    })
+    setBottomSheetVisible(false)
+  }
+
+  const onPressShare = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.external_exchange_qr_share, {
+      flow,
+    })
+    dispatch(shareQRCode(qrSvgRef.current))
+  }
+
+  const onPressCopy = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.external_exchange_qr_copy_address, {
+      flow,
+    })
+    Clipboard.setString(address || '')
```

Comment:
Is the `|| ''` for type safety?

Reply:
Yeah, we can also do `address!` here, but I think I prefer `|| ''` and it's a pattern we use elsewhere as well.